### PR TITLE
Small improvements

### PR DESCRIPTION
**Pull request by lynxis**
Hi,
here are some improvements.
They are tested somehow, but they need additional testing. I would like have some automatic tests :).
I'll use this branch for my wifi at home for a week now, but this wouldn't cover all cases.

Best,
lynxis

### CODE REVIEW

**Comment by mitar** [comment]
Feel free to suggest how to do tests. :-) Travis CI sadly does not allow LXC containers inside, that one could create multiple virtual boxes and networking stack to test things.

File: broker/l2tp_broker.py [hunk 1]
```diff
@@ -59,7 +59,7 @@
   # Message data (with length prefix)
   cs.PascalString("data"),
   # Pad the message so it is at least 12 bytes long
-  cs.Padding(lambda ctx: max(0, 6 - len(ctx["data"]))),
+  cs.Optional(cs.Padding(lambda ctx: max(0, 6 - len(ctx["data"])))),
```
Comment (kostko) [inline broker/l2tp_broker.py]: Can you also remove the code around line 525 (method `Tunnel._run`) that adds padding before parsing if there is none, as now this is probably not necessary and your fix is actually much nicer.

File: broker/l2tp_broker.py [hunk 2]
```diff
@@ -772,7 +773,7 @@ def hook(self, name, *args):
       return
 
     # Execute the registered hook
-    logger.debug("Executing hook '%s' via script '%s'..." % (name, script))
+    logger.debug("Executing hook '%s' via script '%s'... %s" % (name, script, str([str(x) for x in args])))
```
Comment (kostko) [inline broker/l2tp_broker.py]: Can you move the last `%s` to be placed next to the script name? So something like `via script '%s %s'...`.

**Comment by kostko** [comment]
Looks good, just address the above comments and I will merge it. Thanks!

**Comment by mitar** [comment]
@lynxis, please address the pull request feedback so that we can merge this in. ;-)

**Comment by lynxis** [comment]
done.

**Comment by kostko** [comment]
Ok, looks good now, thanks, I will merge it.

> Travis CI sadly does not allow LXC containers inside, that one could create multiple virtual boxes and networking stack to test things.

But CircleCI allows running your own Docker containers ;-) So perhaps some tests could be done this way?

**Comment by mitar** [comment]
> But CircleCI allows running your own Docker containers ;-) So perhaps some tests could be done this way?

Are they free for open source projects?

**Comment by kostko** [comment]
Yes, they even give you more containers so you can run more parallel builds if you are an open source project.

**Pull request merged**